### ISSUE TITLE
Move `#warn` calls from Kernel to Logger

### ISSUE THREAD

**Issue by AlexB52** [guard/listen#572]
I'm looking to silence the Symlink error. 

```
** ERROR: directory is already being watched! **

Directory: %s

is already being watched through: %s

MORE INFO: #{README_URL}
```

People have mentioned you can monkey patch `Listen::Record::SymlinkDetector` class: https://github.com/rails/webpacker/issues/1990

I see that `logger` has a warn level that can be set, but some warn messages are made by Kernel

kernel warn references:
* https://github.com/guard/listen/blob/master/lib/listen/adapter.rb#L39
* https://github.com/guard/listen/blob/master/lib/listen/adapter/bsd.rb#L37
* https://github.com/guard/listen/blob/master/lib/listen/adapter/darwin.rb#L33
* https://github.com/guard/listen/blob/master/lib/listen/adapter/windows.rb#L23
* https://github.com/guard/listen/blob/master/lib/listen/record/symlink_detector.rb#L36

Are there specific reasons for using `Kernel#warn` instead of `Logger#warn`?
Can we move the calls from `Kernel` to `Logger`?

**Comment by LouisaNikita** [comment]
我已收到你的邮件，谢谢！

**Comment by ColinDKelley** [comment]
> Are there specific reasons for using Kernel#warn instead of Logger#warn?

Well, they are different in that `Kernel.warn` will go to `STDERR` vs the log file. So it's probably more likely that `Kernel.warn` will be seen?

I notice the code is using `Kernel.warn` whereas it could equally just call `warn` which would wind up in `Kernel#warn` (the base class method).

I wonder if we should make this warning explicitly disablable?

**Comment by ColinDKelley** [comment]
@AlexB52 How does this approach look? https://github.com/guard/listen/pull/579

**Comment by AlexB52** [comment]
@ColinDKelley 
That went from no flexibility to maximum flexibility that's awesome. 
The configuration to switch from :warn to :log to :silent looks really handy. 
Thank you

**Comment by ColinDKelley** [comment]
@AlexB52 Thanks! That style of configuration was an inspiration from Rails configs that work like that. Let's keep the discussion happening over on that PR.

**Comment by ColinDKelley** [comment]
Resolved in v3.9.0. See [README](https://github.com/guard/listen/blob/untagged-41b10b65a30f00839a7d/README.md#adapter-warnings) for documentation.
